tests/v2/validation/provisioning/rke2/psact_test.go

```diff
@@ -0,0 +1,124 @@
+	"github.com/rancher/shepherd/extensions/provisioninginput"
+	"github.com/rancher/shepherd/extensions/users"
+	password "github.com/rancher/shepherd/extensions/users/passwordgenerator"
+	"github.com/rancher/shepherd/pkg/config"
+	namegen "github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type RKE2PSACTTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	session            *session.Session
+	standardUserClient *rancher.Client
+	provisioningConfig *provisioninginput.Config
+}
+
+func (r *RKE2PSACTTestSuite) TearDownSuite() {
+	r.session.Cleanup()
+}
+
+func (r *RKE2PSACTTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	r.session = testSession
+	r.provisioningConfig = new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, r.provisioningConfig)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(r.T(), err)
+	r.client = client
+
+	r.provisioningConfig.RKE2KubernetesVersions, err = kubernetesversions.Default(
+		r.client, clusters.RKE2ClusterType.String(), r.provisioningConfig.RKE2KubernetesVersions)
+	require.NoError(r.T(), err)
+
+	enabled := true
+	var testuser = namegen.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(r.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(r.T(), err)
+
+	r.standardUserClient = standardUserClient
+}
+
+func (r *RKE2PSACTTestSuite) TestRKE2PSACTNodeDriverCluster() {
+	nodeRolesDedicated := []provisioninginput.MachinePools{provisioninginput.EtcdMachinePool, provisioninginput.ControlPlaneMachinePool, provisioninginput.WorkerMachinePool}
+
+	tests := []struct {
+		name         string
+		machinePools []provisioninginput.MachinePools
+		psact        provisioninginput.PSACT
+		client       *rancher.Client
+	}{
+		{"Rancher Privileged " + provisioninginput.AdminClientName.String(), nodeRolesDedicated, "rancher-privileged", r.client},
+		{"Rancher Privileged " + provisioninginput.StandardClientName.String(), nodeRolesDedicated, "rancher-privileged", r.standardUserClient},
+		{"Rancher Restricted " + provisioninginput.AdminClientName.String(), nodeRolesDedicated, "rancher-restricted", r.client},
+		{"Rancher Restricted " + provisioninginput.StandardClientName.String(), nodeRolesDedicated, "rancher-restricted", r.standardUserClient},
+		{"Rancher Baseline " + provisioninginput.AdminClientName.String(), nodeRolesDedicated, "rancher-baseline", r.client},
+		{"Rancher Baseline " + provisioninginput.StandardClientName.String(), nodeRolesDedicated, "rancher-baseline", r.standardUserClient},
```

Comment:
I know by default there's an empty string case `""`. for posterity should that be added here, or do we just want to say 'null case is tested in general provisioning'?

Reply:
I would say the latter is fine since this test is specifically testing defined PSACT templates.